src/main/java/io/github/bobocodebreskul/context/scan/RecursiveClassPathAnnotatedBeanScanner.java

```diff
@@ -0,0 +1,78 @@
+package io.github.bobocodebreskul.context.scan;
+
+import static io.github.bobocodebreskul.context.support.GeneralConstants.EMPTY;
+import static java.util.Arrays.asList;
+import static java.util.function.Predicate.not;
+import static java.util.stream.Collectors.toSet;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Queue;
+import java.util.Set;
+
+/**
+ * Implementation of {@link ClassPathAnnotatedBeanScanner} to recursively find all bean definitions
+ * in package. Additionally, performs package scan of all found configurations.
+ *
+ * @author Vitalii Katkov
+ */
+public class RecursiveClassPathAnnotatedBeanScanner implements ClassPathAnnotatedBeanScanner {
+
+  private static final String PACKAGE_DELIMITER = ".";
```

Comment:
let's move it to constants too

Reply:
I don't see big reason of moving very specific constant to some global constants class, i need good reason to do that. Also i try to avoid one big class with all constant as it's always mess in the long run, so i would appreciate if you will give more detailed explanation here - why to move very class specific constants out and where to put it?